pkg/preparer/setup.go

```diff
@@ -116,9 +94,9 @@ func New(preparerConfig *PreparerConfig, logger logging.Logger) (*Preparer, erro
 	}
 
 	var err error
-	var keyring openpgp.KeyRing = nil
+	var verifier auth.SigVerifier
 	if preparerConfig.KeyringPath != "" {
-		keyring, err = loadKeyring(preparerConfig.KeyringPath)
+		verifier, err = auth.FileKeyringVerifier(preparerConfig.KeyringPath)
```

Comment:
That's a bit more complex, because the verifier doesn't just return true: it returns the PGP entity of the signer. Grated, the _who_ is usually ignored (no signer? fail!) but in [one case](https://github.com/square/p2/blob/master/pkg/preparer/orchestrate.go#L182) there are additional checks on the signer to implement a hard-coded authorization policy.

I think long-term, authorization and authentication need to be better addressed.  For now, can we leave this?